cylc/flow/task_job_mgr.py

```diff
@@ -265,13 +270,17 @@ def submit_task_jobs(self, workflow, itasks, curve_auth,
         if is_simulation:
             return self._simulation_submit_task_jobs(itasks, workflow)
         # Prepare tasks for job submission
-        prepared_tasks, bad_tasks = self.prep_submit_task_jobs(
-            workflow, itasks)
+        (
+            prepared_tasks,
+            bad_tasks,
+            out_of_hosts_tasks
+        ) = self.prep_submit_task_jobs(workflow, itasks)
         # Reset consumed host selection results
         self.task_remote_mgr.subshell_eval_reset()
 
-        if not prepared_tasks:
+        if not prepared_tasks and not out_of_hosts_tasks:
             return bad_tasks
```

Comment:
⚠ I think this means that `out_of_hosts` below on L364 is now possibly unbound

Reply:
I don't think it will be:

Because `auth_itasks` is set by

```python
        for itask in prepared_tasks:
            platform_name = itask.platform['name']
            auth_itasks.setdefault(platform_name, [])
            auth_itasks[platform_name].append(itask)
```

There will not be items in `auth_itasks` that do not have >=1 value.

This leads to `for itask in itasks:` (line 298) always being run at least once, and that loop always sets `out_of_hosts`.

Reply:
But if `prepared_tasks` is empty, `auth_tasks` will be empty too